recipes/libcusolver/meta.yaml

```
@@ -0,0 +1,181 @@
+{% set cuda_version = "12.0" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set target_name = "x86_64-linux" %}  # [linux64]
+{% set target_name = "ppc64le-linux" %}  # [ppc64le]
+{% set target_name = "sbsa-linux" %}  # [aarch64]
+{% set target_name = "x64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/{{ name }}/{{ platform }}/{{ name }}-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: c9460db500d0c7d14debb2c8e0d88fc26a39197363f1b000ccf760866f636f12  # [linux64]
+  sha256: bf3c9a090652d011aafe2190c6f26fe108194cf8512d3c571e406054740168bb  # [ppc64le]
+  sha256: 0ca46b73520657f96f921711c05af7053bd398328d510aecddfff2521c000e91  # [aarch64]
+  sha256: c09836df1f43a3496f2b61ad8827af662ace298508869564a40c30d909ba550f  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+outputs:
+  - name: libcusolver
+    files:
+      - lib/libcusolver*.so.*                            # [linux]
+      - targets/{{ target_name }}/lib/libcusolver*.so.*  # [linux]
+      - Library\bin\cusolver*.dll                        # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - sysroot_{{ target_platform }} 2.17    # [linux]
+      host:
+        - cuda-version {{ cuda_version }}
+        - libcublas
+        - libnvjitlink
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+        - {{ pin_compatible("libcublas", max_pin="x.x") }}
+        - {{ pin_compatible("libnvjitlink", max_pin="x.x") }}
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    test:
+      commands:
+        - test -L $PREFIX/lib/libcusolver.so.{{ version }}                  # [linux]
+        - test -L $PREFIX/lib/libcusolver.so.{{ version.split(".")[0] }}    # [linux]
+        - test -L $PREFIX/targets/{{ target_name }}/lib/libcusolver.so.{{ version }}                  # [linux]
+        - test -f $PREFIX/targets/{{ target_name }}/lib/libcusolver.so.{{ version.split(".")[0] }}    # [linux]
+        - test -L $PREFIX/lib/libcusolverMg.so.{{ version }}                # [linux]
+        - test -L $PREFIX/lib/libcusolverMg.so.{{ version.split(".")[0] }}  # [linux]
+        - test -L $PREFIX/targets/{{ target_name }}/lib/libcusolverMg.so.{{ version }}                # [linux]
+        - test -f $PREFIX/targets/{{ target_name }}/lib/libcusolverMg.so.{{ version.split(".")[0] }}  # [linux]
+        - if not exist %LIBRARY_BIN%\\cusolver*.dll exit 1                  # [win]
+        - if not exist %LIBRARY_BIN%\\cusolverMg*.dll exit 1                # [win]
```

Tiny suggestion: We discussed this on libcublas and it seems that only one slash is needed here. Let's match that style.
```suggestion
        - if not exist %LIBRARY_BIN%\cusolver*.dll exit 1                  # [win]
        - if not exist %LIBRARY_BIN%\cusolverMg*.dll exit 1                # [win]
```